### ISSUE TITLE
Fine tune a large language model for psychology domain (DOXA AI competition)

### ISSUE THREAD

**Issue by woodthom2** [harmonydata/harmony#55]
## Description

We need to fine tune an LLM for mental health data. We have training and evaluation data available.

We are running a competition on the online platform DOXA AI where you can win up to 500 GBP in vouchers (1st place prize). Check it out here: https://harmonydata.ac.uk/doxa/

## How to get started?

[Create an account on DOXA AI](https://doxaai.com/competition/harmony-matching) and run the example notebook. This will download the training data.

If you would like some tips on how to train an LLM, I recommend [this Hugging Face tutorial](https://huggingface.co/docs/transformers/en/training)

## Rationale

We have noticed for a while that the similarity score that Harmony gives back could be improved. For example, items to do with "sleep" are often grouped together (because of the data that the off the shelf LLMs such as SentenceTransformers are trained on) while a psychologist would consider them to be different. [Read a blog post about how we evaluated the models](https://harmonydata.ac.uk/nlp-semantic-text-matching/measuring-the-performance-of-nlp-algorithms/).

Even the OpenAI models make the same mistakes as the Hugging Face models due to lack of domain understanding. For example, a large number of psychological conditions are sleep related and the LLMs tend to group things together when anything is related to sleep, while a human would consider them entirely different for the point of view of harmonisation of items in a study ("I sleep deeply" vs "I have trouble sleeping" vs "I have little time to sleep"), and similarly "child is bullied" vs "child bullies others" are another false positive. That's why we have the initiative to fine tune an LLM for this domain.

**Comment by pri1712** [comment]
Hey , I am new to Open source but have had prior experience in fine-tuning models in a research setting. Can I try and work on this? I am currently going through how Harmony works, but I would love to try and contribute to this issue.

**Comment by pri1712** [comment]
hey , is this issue still open? If so , I would like to have a go at it!

**Comment by woodthom2** [comment]
Yes it's still open until 10 January. Please follow the link to the DOXA
website and download the training data there

**Comment by woodthom2** [comment]
Hi @pri1712 - happy new year! Did you find the details for the LLM competition? https://doxaai.com/competition/harmony-matching - feel free to also contact me on [Harmony Discord](https://discord.gg/harmonydata) - my name is Thomas Wood

**Comment by woodthom2** [comment]
The competition is closed. Thanks for your entries.

See https://harmonydata.ac.uk/matching-challenge-winner-announcement/

The wait is over! We have now closed the Matching Challenge which was hosted on [DOXA AI](https://doxaai.com/competition/harmony-matching). Over the course of the competition we saw a total of 26 participants with 14 finalists making it onto the [scoreboard](https://doxaai.com/competition/harmony-matching/scoreboard). The final days were tense with many participants improving on their scores, submitting different methods and swapping places at the top of the scoreboard.


**..and the winners are...**


**1st Place 🏅** 
[José Inés Martínez Berard](https://www.linkedin.com/in/jos%C3%A9-in%C3%A9s-mart%C3%ADnez-berard-7b86a9222/) 'End' @oreug, with an impressive MAE (mean absolute error) of 20.348 
Prize: **£500** in Amazon Vouchers 
Congratulations! look out for an email from us with your prize


**2nd Place 🥈** 
[Rafi Ahmed Riyaz Ahmed Patel](https://www.linkedin.com/in/rafi-ahmed-patel-bb2954202) @rafa, with a very close 2nd place MAE of 20.544 
Prize: **£250** in Amazon Vouchers 
Well done 👏👏👏 your prize will be emailed to you soon
